src/v/cluster/health_monitor_frontend.h

```diff
@@ -84,6 +89,11 @@ class health_monitor_frontend
       get_cluster_health_overview(model::timeout_clock::time_point);
 
     ss::future<bool> does_raft0_have_leader();
+    /**
+     * Method validating if a node is known and alive. It will return an empty
+     * optional if the node is not present in node status table.
```

Comment:
> node is not present in node status table

What is the semantics of this? Is it "node is unknown" or "there hasn't been a successful ping yet"? I remember we were struggling with this in partition balancer...

Reply:
there is always possibility that node will be in members table already but wasn't yet pinged.